network/network.go

```diff
@@ -513,9 +514,61 @@ func (n *network) KnownPeers() ([]byte, []byte) {
 	return n.ipTracker.Bloom()
 }
 
-func (n *network) Peers(except ids.NodeID, knownPeers *bloom.ReadFilter, salt []byte) []*ips.ClaimedIPPort {
-	return n.ipTracker.GetGossipableIPs(
-		except,
+// There are 3 types of responses:
+//
+// - Respond with subnet IPs tracked by both ourselves and the peer
+//   - We do not consider ourself to be a primary network validator
+//
+// - Respond with all subnet IPs
+//   - The peer requests all peers
+//   - We believe the peer to be a primary network validator
+//   - We believe ourself to be a primary network validator
+//
+// - Respond with subnet IPs tracked by the peer
+//   - Either the peer does not request all peers or we don't consider them to
+//     be a primary network validator
+//   - We believe ourself to be a primary network validator
+//
+// The reason we allow the peer to request all peers is so that we can avoid
+// sending unnecessary data in the case that we consider them a primary network
+// validator but they do not consider themselves one.
+func (n *network) Peers(
+	peerID ids.NodeID,
+	trackedSubnets set.Set[ids.ID],
+	requestAllPeers bool,
+	knownPeers *bloom.ReadFilter,
+	salt []byte,
+) []*ips.ClaimedIPPort {
+	areWeAValidator := n.config.Validators.GetWeight(constants.PrimaryNetworkID, n.config.MyNodeID) != 0
```

Comment:
I had actually forgotten that in other places we used `GetValidator`... I think `GetWeight` is probably more efficient because it doesn't copy the struct (and can be called inline)... But I think it probably makes sense to standardize it... Do you care which one I switch to?

Reply:
I changed the PR to use `GetValidator` to be consistent with the other calls